configstore.go

```diff
@@ -17,8 +17,11 @@ var (
 
 func init() {
 	RegisterProviderFactory("file", fileProvider)
+	RegisterProviderFactory("file_refresh", fileRefreshProvider)
```

Comment:
I find this kind of weird to define a new protocol for only an optional behaviour. Maybe we can add this as a query parameter? Something like a DSN: `file:///path/to/secrets?refresh=true` ? If not, the scheme could be `file+refresh` which would be more explicit that it is the same protocol with additional behaviour.

Reply:
I guess replacing `_` by `+` is better because in this case, we arent' able to enable refresh on providers which doesn't allow refresh